libbeat/outputs/elasticsearch/client.go

```diff
@@ -185,6 +186,36 @@ func NewClient(
 		proxyURL:         s.Proxy,
 	}
 
+	if s.CacheRedirect {
+		client.Connection.http.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+			if len(via) != 1 {
+				return nil
+			}
+
+			if via[0].Method != "GET" {
+				return http.ErrUseLastResponse
```

Comment:
Why stop processing if method != "GET"? So output client tries to reconnect? In this case I wouldn't use  `ErrUseLastResponse` (which makes the client think no error occured, but check response code only). Returning an response with status 30x might actually lead to data loss if `PublishEvent` is executed.